lib/logidze/utils/check_pending.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require_relative "./function_definitions"
+require_relative "./pending_migration_error"
+
+module Logidze
+  module Utils
+    # This Rack middleware is used to verify that all functions are up to date
+    class CheckPending
+      def initialize(app)
+        @app = app
+        @needs_check = true
+        @mutex = Mutex.new
+      end
+
+      delegate :connection, to: ActiveRecord::Base
+
+      def call(env)
+        @mutex.synchronize do
+          if @needs_check
+            notify_or_raise! if needs_migration?
+          end
+          @needs_check = false
+        end
+
+        @app.call(env)
+      end
+
+      private
+
+      def notify_or_raise!
+        msg = "Logidze needs upgrade. Run `bundle exec rails generate logidze:install --update`"
```

Comment:
Let's make this message more visible, something like this: https://github.com/anycable/anycable-rails/blob/dfb18ee5c665fad15090bbfc2386081fc3ee51c3/lib/anycable/rails.rb#L59

Reply:
Done =)